### PR TITLE
refactor: remove shimmer support

### PR DESCRIPTION
**Pull request by antbern**
# Pull Request Template

## Motivation and Context

Shimmer testnet is deprecated, so we need to switch to using the IOTA testnet for unit-tests and examples.

## Summary

Provide a short summary of changes in this pull request.

- Add a ci job that runs all unit tests (coverage job excludes some wallet-related tests)
- Use Iota wallet instead of Smr for unit tests. Fix code that syncs the wallet to always sync, instead of skipping whenever two syncs happen too close in time.
- use iota for examples and regression tests
- Remove shimmer support completely

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [x] Code follows project style and guidelines.
- [x] No redundant or duplicate code.
- [x] No added new dependencies without clear justification and approval.

### Documentation

- [x] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [x] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [x] Tests pass locally.

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -108,14 +108,38 @@ jobs:
         run: curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
 
       - name: Run unit tests
-        run: cargo llvm-cov nextest --profile ci --locked --lib --workspace --no-fail-fast --all-features --cobertura --output-path cobertura.xml --exclude cryptpay-sdk-jni --exclude cryptpay-sdk-swift --exclude cryptpay-sdk-wasm --ignore-filename-regex "sdk/bindings/|api_types/|.*/error\.rs"
+        run: make unit-tests
```
Comment (kristifidani) [inline .github/workflows/ci.yml:111]: Is there a reason for keeping both the unit tests and coverage jobs? Maybe we can remove unit tests as we did in the backend

**Review by kristifidani — APPROVED** [review]
Looks good 💯 Let's merge 🚀 

Just 1 small comment which is not quite related to the changes 😅

File: .github/workflows/ci.yml (hunk 1 above)
Comment (antbern) [reply .github/workflows/ci.yml:111]: Yes,  some tests are excluded when running the coverage tests (especially those related to wallet actions). So we should keep both to make sure all tests are run and also get the coverage info 🚀  (I actually added the unit-test job in this PR )

**Review by hauju — APPROVED** [review]
